lifecycle.go

```diff
@@ -180,8 +180,8 @@ var defaultLogConsumersHook = func(cfg *LogConsumerConfig) ContainerLifecycleHoo
 	}
 }
 
-// defaultReadinessHook is a hook that will wait for the container to be ready
-var defaultReadinessHook = func() ContainerLifecycleHooks {
+// DefaultReadinessHook is a hook that will wait for the container to be ready
+func DefaultReadinessHook() ContainerLifecycleHooks {
```

Comment:
I don't think we need to expose these variables to the public API in order to run the tests. Do you see any other way to achieve that?

Reply:
The `DefaultSomething` ones I think I might have to keep as exposed because they're used in tests that end up imported as examples, like this one: https://github.com/testcontainers/testcontainers-go/pull/2269/files/2f20304bbcd6ef52418ae330efb70be4af90cef4#diff-8c1de2f5177b67b15485d779c1510b2fa4abeb3cd1e82c18d43d0c2e7871ed19R74. If we keep that test in the internal package, that would mean losing the `testcontainer` prefix 🤔

Reply:
After some shuffling around, I managed to rename them 😄